Packages/blueberry-jam-core/Runtime/LevelLoader/SceneTransitionerHelper.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections;
+using Codice.Client.BaseCommands.TubeClient;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BJ
+{
+    internal class SceneTransitionerHelper : SingletonGameObject<SceneTransitionerHelper>
+    {
+        [Header("Game Object References")]
+        [SerializeField] private CanvasGroup Crossfade;
+        [SerializeField] private Image LoadingIcon;
+        [Header("Animation Reference Storage")]
+        [SerializeField] private Sprite[] Frames;
+
+        [Header("Animation Configuration")]
+        [SerializeField] private float TransitionTime = 1f;
+        [SerializeField] private AnimationCurve EaseTransitionCurve;
+
+        /* ----- Helper Variables -----*/
+        private float StartTransitionTime;
+
+        protected override void Awake()
+        {
+            base.Awake();
```

Comment:
I am trying to ensure that the Singleton's Awake method is being called with this line of code. Is it redundant? Will the singleton's Awake method be called without this?

Reply:
You do need to have this line of code to guarantee that. Otherwise a base method will be ignored if it is overridden.